_pytest/main.py

```diff
@@ -208,6 +210,25 @@ def pytest_ignore_collect(path, config):
     return False
 
 
+def pytest_collection_modifyitems(items, config):
+    deselectopt = config.getoption("deselect")
+    if deselectopt is None:
+        return
+
+    remaining = []
+    deselected = []
+    for colitem in items:
+        for opt in deselectopt:
+            if colitem.nodeid.startswith(opt):
```

Comment:
startswith can take a tuple which can be used as a much better selector

instead of a loop and the for else hack it could simply use `colitem.nodeid.startswith(deselect_prefixes)`

i presume the break was not added due to the expectation of the prefixes being distinct/non-overlapping

Reply:
Oh cool, thanks I didnt realize `str.startswith()` takes a tuple.  That's great.